components/core/include/core/platform/context.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright (c) 2023, Thomas Atkinson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string_view>
+#include <vector>
+
+namespace vkb
+{
+/**
+ * @brief A platform context contains abstract platform specific operations
+ *
+ *        A platform can be thought as the physical device and operating system configuration that the application
+ *        is running on.
+ *
+ *        Some platforms can be reused across different hardware configurations, such as Linux and Macos as both
+ *        are POSIX compliant. However, some platforms are more specific such as Android and Windows
+ */
+class PlatformContext
+{
+  public:
+	PlatformContext()          = default;
+	virtual ~PlatformContext() = default;
+
+	virtual std::vector<std::string_view> arguments() const = 0;
+
+	virtual std::string_view external_storage_directory() const = 0;
+
+	virtual std::string_view temp_directory() const = 0;
```

Comment:
Why do you make those three functions pure virtual? Do you expect to have any platform that has that data in a different form?
And why do you return a `std::vector<std::string_view>`, instead of just a `const std::vector<std::string>&`? You're approach probably requires some preparation in every arguments() implementation (if the implementation stays in the derived classes).

Reply:
I think i made them pure virtual  due to android being the outlier with its `static std:;string` usage

I have kept them virtual just incase there are other platforms in the future which need to change a few things. Also moved to `const&` as suggested. Changes are inbound just building and testing now